### PR TITLE
[QA-458]: Update response time changes in remaining scripts

### PR DESCRIPTION
**Pull request by abhinaysrivastavaperf**
### What?
Response time percentiles added for remaining scripts based on the work done in #528 

#### Changes:
Response time percentile changes updated in below scripts
- OLH
- Bravo (VC Storage and AIS)
- CIMIT
- Lime
- Orange
- IPV Core

---

### Why?
To ensure response time percentiles are recorded in the k6 end of test summary report.

---

### Related:
- #528

### CODE REVIEW

**Review by TaherKapasi — APPROVED** [review]

File: deploy/scripts/src/cri-lime/test.ts [hunk 1]
```diff
@@ -163,13 +165,33 @@ const profiles: ProfileList = {
...
+    'B01_Fraud_02_ContinueToCheckFraudDetails',
+    'B01_Fraud_02_ContinueToCheckFraudDetails::01_CRICall',
+    'B01_Fraud_02_ContinueToCheckFraudDetails::02_CoreStubCall'
+  ],
+  drivingLicence: [
+    'B02_Driving_01_DLEntryFromCoreStub',
+    `B02_Driving_02_Select${licenceIssuer}`,
+    `B02_Driving_03_${licenceIssuer}_EnterDetailsConfirm`,
+    `B02_Driving_03_${licenceIssuer}_EnterDetailsConfirm::01_CRICall`,
+    `B02_Driving_03_${licenceIssuer}_EnterDetailsConfirm::02_CoreStubCall`
```
Comment (Tom-Dann) [inline deploy/scripts/src/cri-lime/test.ts:182]: This is only going to generate thresholds for one of `DVA` or `DVLA`, it needs to be both

File: deploy/scripts/src/ipv-core/test.ts [hunk 2]
```diff
@@ -227,220 +301,203 @@ const stubCreds = {
...
+      { isStatusCode302 }))
+      res = group(groups[30].split('::')[1], () => timeRequest(() => // 02_OrchStub
+        http.get(res.headers.Location, {
+          headers: { Authorization: `Basic ${encodedCredentials}` }
+        }),
+      { isStatusCode200, ...pageContentCheck('User information') }))
+    }, {})
   })
+  iterationsCompleted.add(1)
+  console.log(userId + ',' + testEmail)
```
Comment (Tom-Dann) [inline deploy/scripts/src/ipv-core/test.ts:496]: Why has this debug statement been added?

File: deploy/scripts/src/accounts/test.ts [hunk 3]
```diff
@@ -216,459 +267,365 @@ const phoneData = {
...
   const newEmail = `perftest${timestamp}${iteration}@digital.cabinet-office.gov.uk`
   iterationsStarted.add(1)
 
-  res = group('B01_ChangeEmail_01_LaunchAccountsHome GET', () =>
-    timeRequest(() => http.get(env.envURL, {
-      tags: { name: 'B01_ChangeEmail_01_LaunchAccountsHome' }
-    }),
-    { isStatusCode200, ...pageContentCheck('Services you can use with GOV.UK One Login') }))
+  res = group(groups[0], () => timeRequest(() => // B01_ChangeEmail_01_LaunchAccountsHome
+    http.get(env.envURL), { isStatusCode200, ...pageContentCheck('Services you can use with GOV.UK One Login') }))
```
Comment (Tom-Dann) [inline deploy/scripts/src/accounts/test.ts:278]:
```suggestion
    http.get(env.envURL),
    { isStatusCode200, ...pageContentCheck('Services you can use with GOV.UK One Login') }
  ))
```
Suggestion: Don't inline all the calls on this one line

File: deploy/scripts/src/accounts/test.ts [hunk 4]
```diff
@@ -216,459 +267,365 @@ const phoneData = {
...
-    timeRequest(() => res.submitForm({
-      params: { tags: { name: 'B05_ValidateUser_02_ClickSignIn' } }
-    }), { isStatusCode200, 'verify page content': r => (r.body as string).includes('Enter your email address to sign in to your GOV.UK One Login') }))
+  res = group(groups[1], () => timeRequest(() => // B05_ValidateUser_02_ClickSignIn
+    res.submitForm({
+      fields: {
+        supportInternationalNumbers: 'true'
+      }
+    }),
+  { isStatusCode200, 'verify page content': r => (r.body as string).includes('Enter your email address to sign in to your GOV.UK One Login') }))
```
Comment (Tom-Dann) [inline deploy/scripts/src/accounts/test.ts:541]:
```suggestion
  { isStatusCode200, ...pageContentCheck('Enter your email address to sign in to your GOV.UK One Login') }))
```
Not necessarily in the scope of this PR, but just noticed this scenario is not using the `pageContentCheck()` utility function for most of the assertions.

This change can go in another PR, but just noting it here whilst reviewing these lines of code

**Review by Tom-Dann — CHANGES_REQUESTED** [review]

File: deploy/scripts/src/ipv-core/test.ts (hunk 2 above)
Comment (abhinaysrivastavaperf) [reply deploy/scripts/src/ipv-core/test.ts:496]: This was added to validate the ID reuse script and has been removed now

File: deploy/scripts/src/accounts/test.ts (hunk 3 above)
Comment (abhinaysrivastavaperf) [reply deploy/scripts/src/accounts/test.ts:278]: Updated

File: deploy/scripts/src/accounts/test.ts (hunk 4 above)
Comment (abhinaysrivastavaperf) [reply deploy/scripts/src/accounts/test.ts:541]: Thanks. This will be updated in the next PR. Leaving it as is for this PR

File: deploy/scripts/src/cri-lime/test.ts (hunk 1 above)
Comment (abhinaysrivastavaperf) [reply deploy/scripts/src/cri-lime/test.ts:182]: Yeah did not realise that I moved this in the init stage instead of the virtual user code section. This has now been updated accordingly.

File: deploy/scripts/src/cri-lime/test.ts [hunk 5]
```diff
@@ -435,16 +430,47 @@ export function drivingLicence (): void {
...
+      timeRequest(() => {
+        res = group(groups[7].split('::')[1], () => timeRequest(() => // 01_CRICall
+          res.submitForm({ fields, params: { redirects: 2 }, submitSelector: '#continue' }), { isStatusCode302 }))
+        res = group(groups[8].split('::')[1], () => timeRequest(() => // 02_CoreStubCall
+          http.get(res.headers.Location, { headers: { Authorization: `Basic ${encodedCredentials}` } }),
+        { isStatusCode200, ...pageContentCheck('Verifiable Credentials') }))
+      }, {})
+    })
+    iterationsCompleted.add(1)
+  }
```
Comment (Tom-Dann) [inline deploy/scripts/src/cri-lime/test.ts:473]:
```suggestion
  } else if (licenceIssuer === 'DVLA') {
    res = group(groups[5], () => timeRequest(() => // B02_Driving_02_Select_DVLA
      res.submitForm({
        fields: { licenceIssuer },
        submitSelector: '#submitButton'
      }),
    { isStatusCode200, ...pageContentCheck('Enter your details exactly as they appear on your UK driving licence') }))

    sleepBetween(1, 3)

    group(groups[6], () => { // B02_Driving_03_EnterDetailsConfirm_DVLA
      timeRequest(() => {
        res = group(groups[7].split('::')[1], () => timeRequest(() => // 01_CRICall
          res.submitForm({ fields, params: { redirects: 2 }, submitSelector: '#continue' }), { isStatusCode302 }))
        res = group(groups[8].split('::')[1], () => timeRequest(() => // 02_CoreStubCall
          http.get(res.headers.Location, { headers: { Authorization: `Basic ${encodedCredentials}` } }),
        { isStatusCode200, ...pageContentCheck('Verifiable Credentials') }))
      }, {})
    })
  }
  iterationsCompleted.add(1)
```
Keep `iterationsCompleted` as only called once, at the end of the scenario

File: deploy/scripts/src/cri-lime/test.ts
Comment (Tom-Dann) [inline deploy/scripts/src/cri-lime/test.ts:1]: This change is adding code duplication. You should be able to keep the existing pattern of reusing steps (using a variable for `DVA` or `DVLA`), whilst implementing the thresholds work.

**Review by Tom-Dann — CHANGES_REQUESTED** [review]

File: deploy/scripts/src/cri-lime/test.ts
Comment (abhinaysrivastavaperf) [reply deploy/scripts/src/cri-lime/test.ts:1]: Updated in the next commit

File: deploy/scripts/src/cri-lime/test.ts (hunk 5 above)
Comment (abhinaysrivastavaperf) [reply deploy/scripts/src/cri-lime/test.ts]: Updated

File: deploy/scripts/src/cri-lime/test.ts [hunk 6]
```diff
@@ -228,61 +253,44 @@ export function fraud (): void {
...
   })
   iterationsCompleted.add(1)
 }
 
 export function drivingLicence (): void {
-  let res: Response
   type drivingLicenceIssuer = 'DVA' | 'DVLA'
   const licenceIssuer: drivingLicenceIssuer = (Math.random() <= 0.5) ? 'DVA' : 'DVLA'
+  let groups: string[]
+  licenceIssuer === 'DVA' ? groups = groupMap.drivingLicence.filter((issuer) => issuer.includes('DVA')) : groups = groupMap.drivingLicence.filter((issuer) => issuer.includes('DVLA'))
```
Comment (Tom-Dann) [inline deploy/scripts/src/cri-lime/test.ts:279]:
```suggestion
  let groups = groupMap.drivingLicence.filter(s => s.includes(licenceIssuer))
```
This can be simplified

Comment (abhinaysrivastavaperf) [reply deploy/scripts/src/cri-lime/test.ts]: Thanks, this has been updated.

**Review by Tom-Dann — CHANGES_REQUESTED** [review]
Can you add  `tags: { name: '' }` to the options for each of the scripts just to ensure the `name` value isn't being set now that it has been removed from the params

As per the change in #550 to the authentication script

**Review by Tom-Dann — APPROVED** [review]
